pavelib/assets.py

```diff
@@ -252,24 +252,43 @@ def get_watcher_dirs(theme_dirs=None, themes=None):
             '/edx/app/edxapp/edx-platform/themes/red-theme/cms/static/sass/partials',
         ]
 
+        >> get_watcher_dirs('/edx/app/edx-platform/themes', ['red-theme'], lms)
+        [
+            'common/static',
+            'common/static/sass',
+            'lms/static/sass',
+            'lms/static/sass/partials',
+            '/edx/app/edxapp/edx-platform/themes/red-theme/lms/static/sass',
+            '/edx/app/edxapp/edx-platform/themes/red-theme/lms/static/sass/partials',
+        ]
+
     Parameters:
         theme_dirs (list): list of theme base directories.
-        themes (list): list containing names of themes
+        themes (list): list containing names of themes.
+        system (str): the system for which the assets are watched.
     Returns:
         (list): dirs that need to be added to sass watchers.
     """
     dirs = []
     dirs.extend(COMMON_LOOKUP_PATHS)
+    systems = list(system) if system else ALL_SYSTEMS
+
     if theme_dirs and themes:
         # Register sass watchers for all the given themes
         themes = get_theme_paths(themes=themes, theme_dirs=theme_dirs)
         for theme in themes:
-            for _dir in get_sass_directories('lms', theme) + get_sass_directories('cms', theme):
+            system_dirs = []
+            for sys in systems:
+                system_dirs.extend(get_sass_directories(sys, theme))
+            for _dir in system_dirs:
                 dirs.append(_dir['sass_source_dir'])
                 dirs.extend(_dir['lookup_paths'])
 
-    # Register sass watchers for lms and cms
-    for _dir in get_sass_directories('lms') + get_sass_directories('cms') + get_common_sass_directories():
+    # Register sass watchers for systems
+    system_dirs = []
+    for sys in systems:
+            system_dirs.extend(get_sass_directories(sys))
```

Comment:
This is a plausible spot for a list comprehension:

    system_dirs = [get_sass_directories(sys) for sys in systems]

But that's certainly not a requirement